internal/controller/pipelinerollout/pipelinerollout_controller.go

```diff
@@ -360,46 +360,100 @@ func (r *PipelineRolloutReconciler) reconcile(
 		controllerutil.AddFinalizer(pipelineRollout, common.FinalizerName)
 	}
 
-	newPipelineDef, err := r.makePromotedPipelineDefinition(ctx, pipelineRollout)
+	// check if there's a promoted pipeline yet
+	promotedPipelines, err := ctlrcommon.FindChildrenOfUpgradeState(ctx, pipelineRollout, common.LabelValueUpgradePromoted, nil, false, r.client)
+	if err != nil {
+		return 0, nil, fmt.Errorf("error looking for promoted pipeline: %v", err)
+	}
+	newPipelineDef, err := r.makeTargetPipelineDefinition(ctx, pipelineRollout)
 	if err != nil {
 		return 0, nil, err
 	}
 
-	// Get the object to see if it exists
-	existingPipelineDef, err := kubernetes.GetResource(ctx, r.client, newPipelineDef.GroupVersionKind(),
-		k8stypes.NamespacedName{Name: newPipelineDef.GetName(), Namespace: newPipelineDef.GetNamespace()})
-	if err != nil {
-		// create object as it doesn't exist
-		if apierrors.IsNotFound(err) {
+	requeueDelay := time.Duration(0)
+
+	var existingPipelineDef *unstructured.Unstructured
+
+	if newPipelineDef == nil {
+		requeueDelay = common.DefaultRequeueDelay
+	} else {
+
+		if promotedPipelines == nil || len(promotedPipelines.Items) == 0 {
+
 			numaLogger.Debugf("Pipeline %s/%s doesn't exist so creating", pipelineRollout.Namespace, pipelineRollout.Name)
 			pipelineRollout.Status.MarkPending()
 
+			// need to know if the pipeline needs to be created with "desiredPhase" = "Paused" or not
+			// (i.e. if isbsvc or numaflow controller is requesting pause)
+			userPreferredStrategy, err := usde.GetUserStrategy(ctx, newPipelineDef.GetNamespace())
+			if err != nil {
+				return 0, nil, err
+			}
+			if userPreferredStrategy == config.PPNDStrategyID {
+				needsPaused, _, err := r.shouldBePaused(ctx, pipelineRollout, nil, newPipelineDef, false)
+				if err != nil {
+					return 0, nil, err
+				}
+				if needsPaused != nil && *needsPaused {
+					err = numaflowtypes.PipelineWithDesiredPhase(newPipelineDef, "Paused")
+					if err != nil {
+						return 0, nil, err
+					}
+				}
+
+			}
```

Comment:
Now it's possible to be in the middle of PPND up where we're creating the Pipeline (PPND user strategy with delete/recreate operation)